prover/src/channel.rs

```diff
@@ -137,18 +137,18 @@ where
     /// constraint composition polynomials should be queried.
     ///
     /// The positions are drawn from the public coin uniformly at random.
-    pub fn get_query_positions(&mut self) -> Vec<usize> {
+    pub fn get_query_positions(&mut self, nonce: u64) -> Vec<usize> {
         let num_queries = self.context.options().num_queries();
         let lde_domain_size = self.context.lde_domain_size();
         self.public_coin
-            .draw_integers(num_queries, lde_domain_size)
+            .draw_integers(num_queries, lde_domain_size, nonce)
             .expect("failed to draw query position")
     }
```

Comment:
I wonder if we actually need to pass in the `nonce` as a parameter here. It seems like we should be able to get this from `self` as `self.pow_nonce`. Or does this complicate something?

Reply:
Makes total sense, we can indeed use `self.pow_nonce` to simplify things. I have done that now.